### DISCUSSION TITLE
KafkaTopic does not have status object

### DISCUSSION THREAD

**Discussion by oddiiie** [strimzi/strimzi-kafka-operator#5972] [Q&A]
I'm using Strimzi to generate KafkaTopic, but some of them have status object and some don't.
I'd like to know the conditions under which the status object is given.

* Topic has status object
```
$ kubectl get kafkatopic topic1 -oyaml
apiVersion: kafka.strimzi.io/v1beta1
kind: KafkaTopic
...
spec:
  config:
    max.message.bytes: "2097152"
    message.format.version: 2.4-IV1
  partitions: 1
  replicas: 1
  topicName: topic1
status:
  conditions:
  - lastTransitionTime: "2021-11-29T04:40:03.418Z"
    status: "True"
    type: Ready
  observedGeneration: 2
```

* Topic does not have status object
```
$ kubectl get kafkatopic -oyaml topic2
apiVersion: kafka.strimzi.io/v1beta1
kind: KafkaTopic
...
spec:
  config:
    message.format.version: 2.4-IV1
  partitions: 1
  replicas: 1
  topicName: topic2
```

**Comment by sknot-rh** [comment]
I think this is the same issue as https://github.com/strimzi/strimzi-kafka-operator/issues/5943
Can you please double-check that the topics without the status are not present in Kafka?

**Reply by oddiiie** [reply]
@sknot-rh 
Thank you so much for quick reply.
I'll check it.

**Reply by oddiiie** [reply]
@sknot-rh 
> Can you please double-check that the topics without the status are not present in Kafka?

I checked the Kafka situation.
My KafkaTopic status is not the same as #5943.

The KafkaTopic present in Kafka cluster.

```bash
$ kubectl exec kafka-cluster-v2-kafka-0 -- ./bin/kafka-topics.sh --list --bootstrap-server :9092
...
topic2
```

KafkaTopic  has messages and consumed.

```bash
$ k exec kafka-cluster-v2-kafka-0 -- ./bin/kafka-consumer-groups.sh --describe --bootstrap-server localhost:9092 --all-
GROUP           TOPIC    PARTITION  CURRENT-OFFSET  LOG-END-OFFSET  LAG             CONSUMER-ID                                             HOST            CLIENT-ID
group01       topic2       0          7               7               0               kafka-python-2.0.1-fc66fffb-da0d-44c2-a3f3-1f40b7e0d17a /10.240.0.247   kafka-python-2.0.1
```

But the KafkaTopic does not have status object and events.

```bash
$ kubectl describe kafkatopics.kafka.strimzi.io topic2
...
Spec:
  Config:
    message.format.version:  2.4-IV1
  Partitions:                1
  Replicas:                  1
  Topic Name:                topic2
Events:                      <none>
```

**Comment by scholzj** [comment]
Did you created the KafkaTopic resource? If yes, you should double check that you have the `strimzi.io/cluster` label et correctly, that it is in the same namespace etc. If all these are correct, you should check the Topic Operator logs if there are any issues.

**Reply by oddiiie** [reply]
@scholzj 
Thank you for your support !!

I created the KafkaTopic resource.
And The KafkaTopic has label `strimzi.io/cluster` correctly.

```bash
$ k get kafkatopics.kafka.strimzi.io --show-labels
NAME                                                                    PARTITIONS   REPLICATION FACTOR   LABELS
...
topic2                                                                   1            1                    app.kubernetes.io/instance=topic2,app.kubernetes.io/managed-by=strimzi-kafka-topic-operator,app.kubernetes.io/name=strimzi,argocd.argoproj.io/instance=[helm chartname],strimzi.io/cluster=kafka-cluster-v2
```

I checked The Topic Operator logs but there are no ERROR logs like that. (only INFO logs)

```
[2021-12-01 10:35:02,244] INFO  <opicOperator:473> [oop-thread-0] 53806|periodic kafka topic2|130464615: Reconciling topic topic2, k8sTopic:nonnull, kafkaTopic:nonnull, privateTopic:nonnull
[2021-12-01 10:35:02,244] INFO  <opicOperator:621> [oop-thread-0] 53806|periodic kafka topic2 |130464615: All three topics are identical
[2021-12-01 10:35:02,244] INFO  <opicOperator:1302> [oop-thread-0] Success reconciling KafkaTopic [namespace]/topic2---d96689d6fb70bb081624691070bf2a8431819263
``

**Reply by scholzj** [reply]
From the log it looks like the `topic2` is mapped to KafkaTopic resource `topic2---d96689d6fb70bb081624691070bf2a8431819263`.  I'm not completely sure why.

@sknot-rh Didn't we had some other issue similar to this (not #5943) where the KafkaTopic existied twice for given Kafka topic?

**Reply by sknot-rh** [reply]
I do not recall of any :(

**Reply by scholzj** [reply]
I guess #5674 was what I meant. There were also two KafkaTopic resources for one Kafka topic and it did not worked.

**Reply by sknot-rh** [reply]
Hmm, @dubs11k what version of strimzi do you use?

**Reply by oddiiie** [reply]
@sknot-rh I am using version 0.17.0.
